src/core/net/ip6.cpp

```diff
@@ -1054,6 +1054,15 @@ Error Ip6::ProcessReceiveCallback(Message &          aMessage,
         }
     }
 
+    if (aIpProto == kProtoFragment && mIcmp.ShouldHandleEchoRequest(aMessageInfo))
+    {
+        FragmentHeader fragmentHeader;
+        IgnoreError(aMessage.Read(aMessage.GetOffset(), fragmentHeader));
+
+        // do not pass fragments of ICMP Echo Request messages
+        VerifyOrExit(fragmentHeader.GetNextHeader() != kProtoIcmp6, error = kErrorDrop);
```

Comment:
I think this change here can potentially drop other ICMP6 messages and not just Echo Requests. 

When we get here, we know it is fragment, and that the next header is IMCP6, but it may be any ICMP6 msg (I guess we need to check that it is echo request).

Reply:
In regards to ICMPv6 I think that only echo request and echo reply can be fragmented and both need to be considered here. Isn't it true?

Reply:
Maybe that could be clarified with a comment.